engines/docker/sandbox.go

```diff
@@ -0,0 +1,156 @@
+package dockerengine
+
+import (
+	"context"
+	"time"
+
+	docker "github.com/fsouza/go-dockerclient"
+	"github.com/taskcluster/taskcluster-worker/engines"
+	"github.com/taskcluster/taskcluster-worker/runtime"
+	"github.com/taskcluster/taskcluster-worker/runtime/atomics"
+	"github.com/taskcluster/taskcluster-worker/runtime/caching"
+	"github.com/taskcluster/taskcluster-worker/runtime/ioext"
+)
+
+const dockerEngineKillTimeout = 5 * time.Second
+
+type sandbox struct {
+	engines.SandboxBase
+	monitor     runtime.Monitor
+	containerID string
+	resultSet   engines.ResultSet
+	resultErr   error
+	abortErr    error
+	tempStorage runtime.TemporaryStorage
+	resolve     atomics.Once
+	client      *docker.Client
```

Comment:
Instead of holding a reference to `client` let's just add `engine` as a property... both here and in `sandboxBuilder`.

That way you'll also have access to `environment` if we ever need it.. And similarly, if we add new properties in the future.